lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,22 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_data = @data[:client_cert] || @data[:certificate_path]
-      private_key_data = @data[:client_key] || @data[:private_key_path]
-      private_key_pass = @data[:client_key_pass] || @data[:private_key_pass]
+      client_cert_data = if cert_data = @data[:client_cert_data]
+                           cert_data
+                         elsif path = (@data[:client_cert] || @data[:certificate_path])
+                           File.read path
+                         end
+
+      private_key_data = if key_data = @data[:private_key_data]
```

Comment:
I realize now that the code here makes this unclear due to the local variable names, we should probably change those also (I think they just got left behind). `client_...` is what is documented, so we should standardize on that I think.